app-modules/application/src/Filament/Resources/ApplicationResource/Pages/Concerns/HasSharedFormConfiguration.php

```diff
@@ -0,0 +1,231 @@
+                if ($component->isDisabled()) {
+                    return;
+                }
+
+                $application = $record instanceof Application ? $record : $record->submissible;
+                $applicationStep = $record instanceof ApplicationStep ? $record : null;
+
+                ApplicationField::query()
+                    ->whereBelongsTo($application, 'submissible')
+                    ->when($applicationStep, fn (EloquentBuilder $query) => $query->whereBelongsTo($applicationStep, 'step'))
+                    ->delete();
+
+                $content = $component->decodeBlocksBeforeSave($component->getJSON(decoded: true));
+                $content['content'] = $this->saveFieldsFromComponents(
+                    $application,
+                    $content['content'] ?? [],
+                    $applicationStep,
+                );
+
+                $record->content = $content;
+                $record->save();
+            })
+            ->dehydrated(false)
+            ->columnSpanFull()
+            ->extraInputAttributes(['style' => 'min-height: 12rem;']);
+    }
+
+    public function saveFieldsFromComponents(Application $application, array $components, ?ApplicationStep $applicationStep): array
+    {
+        foreach ($components as $componentKey => $component) {
+            if (array_key_exists('content', $component)) {
+                $components[$componentKey]['content'] = $this->saveFieldsFromComponents($application, $component['content'], $applicationStep);
+
+                continue;
+            }
+
+            if ($component['type'] !== 'tiptapBlock') {
+                continue;
+            }
+
+            $componentAttributes = $component['attrs'] ?? [];
+
+            if (array_key_exists('id', $componentAttributes)) {
+                $id = $componentAttributes['id'] ?? null;
+                unset($componentAttributes['id']);
+            }
+
+            if (array_key_exists('label', $componentAttributes['data'])) {
+                $label = $componentAttributes['data']['label'] ?? null;
+                unset($componentAttributes['data']['label']);
+            }
+
+            if (array_key_exists('isRequired', $componentAttributes['data'])) {
+                $isRequired = $componentAttributes['data']['isRequired'] ?? null;
+                unset($componentAttributes['data']['isRequired']);
+            }
+
+            /** @var ApplicationField $field */
+            $field = $application->fields()->findOrNew($id ?? null);
+            $field->step()->associate($applicationStep);
+            $field->label = $label ?? $componentAttributes['type'];
+            $field->is_required = $isRequired ?? false;
+            $field->type = $componentAttributes['type'];
+            $field->config = $componentAttributes['data'];
+            $field->save();
+
+            $components[$componentKey]['attrs']['id'] = $field->id;
+        }
+
+        return $components;
+    }
```

Comment:
I think this could be shared between modules potentially

Reply:
I think that the configuration of Form and Application will start to vary more and more. It already does so as Application does not have a toggle for "authenticated" because they are authenticated by default.

So leaving them as separate I feel, may be in our best interest.